web/Views/Landlord/Profile.cshtml

```diff
@@ -133,7 +133,7 @@
 
 @if (Model.NumOfProperties > 0)
 {
-    <a asp-action="@(User.IsInRole("Landlord") ? "ViewMyProperties" : "ViewProperties")" asp-controller="Landlord" class="btn btn-primary my-3">View properties</a>
+    <a asp-action="@(User.IsInRole("Admin") ? "ViewProperties" : "ViewMyProperties")" asp-controller="Landlord" asp-route-id="@(User.IsInRole("Admin") ? Model.LandlordId : "")" class="btn btn-primary my-3">View properties</a>

```

Comment:
As an Admin would this not take me to the View Properties for myself if I was also a Landlord?

Reply:
I don't think so - if you're an admin, then it goes to `ViewProperties`, if you're not an admin then it goes to `ViewMyProperties`. So if you're both then it should do the latter.

Reply:
Not sure what you mean. If you are an admin, and on your own page, it will still go to `ViewProperties`